src/hotspot/share/gc/x/xUnload.cpp

```diff
@@ -75,7 +75,7 @@ class XIsUnloadingOopClosure : public OopClosure {
 
 class XIsUnloadingBehaviour : public IsUnloadingBehaviour {
 public:
-  virtual bool has_dead_oop(CompiledMethod* method) const {
+  virtual bool has_dead_oop(nmethod* method) const {
```

Comment:
This now takes an `nmethod` argument, but still calls as_nmethod(). I think that should be removed from this, and all similar functions here in the GC code. If you want, I can do that as a follow-up RFE.

Reply:
I decided to fix it in these changes.